Add missing SLOs to explanation (#212)
appuio/openshift4-docs: docs/modules/ROOT/pages/explanations/slos.adoc

NOTE: As a side effect it also measures out-bound connection issues, which shouldn't be a part of an ingress SLO.
However, the alternative of using an external probe source also measures issues that shouldn't be part of the SLO, so we chose the in-cluster probe source for simplicity.

== Kubernetes API

The Kubernetes API is the main way users interact with the cluster itself.

> Without it users can't change configuration or run new workloads and existing deployments will quickly degrade.

Reviewer: ```suggestion
If the API is not available, users can't change configuration or run new workloads and existing deployments will quickly degrade.
```

A misbehaving Kubernetes API directly impacts the service level.

=== Request Error Rate

****
*99.9% of all requests to the Kubernetes API server succeed or are invalid*
****

This is measured directly at the API server through the following metrics.

[source,promql]
----
# The number of failed valid API requests
apiserver_request_total{code=~"(5..|429)"}

# All API requests
apiserver_request_total
----

> NOTE: We explicitly only look for 500 errors, which indicate a server side error, or at error 429, which indicate that our API server is overloaded.

Reviewer: ```suggestion
NOTE: We only look for HTTP 5xx errors, which indicate a server side error, and HTTP error 429, which indicates that the API server is overloaded.
```

=== Uptime

****
*99.9% of all HTTP probes to the Kubernetes API server succeed*
****

Probes are sent every 10 seconds from a blackbox exporter inside the cluster to the readiness endpoint of the Kubernetes API server.

> This SLI approximates a user's ability to reach at least one API server instance and the API servers uptime.

Reviewer: ```suggestion
This SLI approximates a user's ability to reach at least one API server instance and the API server's uptime.
```

This SLI approximates the user experience of interacting with PVs and PVCs.
It doesn't measure any performance issues with the underlying storage.

== Cluster Network

Reliable cluster networking is essential for nearly every workload.

> Without it users can't reliably access their workload and even moderate packet loss can negatively impact deployments such as databases.

Reviewer: ```suggestion
Without it, users can't reliably access their workload and even moderate packet loss can negatively impact deployments such as databases.
```
